src/D2L.Bmx/Okta/Models/AuthenticateResponse.cs

```diff
@@ -21,17 +21,68 @@ internal record AuthenticateResponseEmbedded(
 	OktaMfaFactor[]? Factors
 );
 
+[JsonPolymorphic(
+	TypeDiscriminatorPropertyName = "factorType",
+	IgnoreUnrecognizedTypeDiscriminators = true
+)]
+[JsonDerivedType( typeof( OktaMfaQuestionFactor ), "question" )]
+[JsonDerivedType( typeof( OktaMfaHardwareToken ), "token" )]
+[JsonDerivedType( typeof( OktaMfaHardwareTokenFactor ), "token:hardware" )]
+[JsonDerivedType( typeof( OktaMfaSoftwareTotpFactor ), "token:software:totp" )]
+[JsonDerivedType( typeof( OktaMfaHotpFactor ), "token:hotp" )]
+[JsonDerivedType( typeof( OktaMfaSmsFactor ), "sms" )]
+[JsonDerivedType( typeof( OktaMfaCallFactor ), "call" )]
+[JsonDerivedType( typeof( OktaMfaEmailFactor ), "email" )]
+internal record OktaMfaFactor {
+	public required string Id { get; set; }
+	public required string Provider { get; set; }
+	public required string VendorName { get; set; }
+
+	[JsonIgnore]
+	public virtual string FactorName => "Unknown";
+	[JsonIgnore]
+	public virtual bool RequireChallengeIssue => false;
+}
+
+internal record OktaMfaQuestionFactor() : OktaMfaFactor() {
+	public override string FactorName => "Security Question";
+	public required OktaMfaQuestionProfile Profile { get; set; }
```

Comment:
does this need to have a setter?

Reply:
Don't think so. This could probably a constructor parameter.